DuckDuckGo/HomeViewController.swift

```diff
@@ -26,110 +26,112 @@ class HomeViewController: UIViewController {
     private struct Constants {
         static let animationDuration = 0.25
     }
-    
-    @IBOutlet weak var passiveContent: UIView!
-    @IBOutlet weak var searchBar: UIView!
-    @IBOutlet weak var searchBarContent: UIView!
-    @IBOutlet weak var searchImage: UIImageView!
-    @IBOutlet weak var searchText: UILabel!
-    
+
+    @IBOutlet weak var bottomConstraint: NSLayoutConstraint!
+    @IBOutlet weak var infoViewHeight: NSLayoutConstraint!
+    @IBOutlet weak var infoView: UIView!
+
     weak var delegate: HomeControllerDelegate?
     weak var chromeDelegate: BrowserChromeDelegate?
-    private var active = false
-    
-    static func loadFromStoryboard(active: Bool) -> HomeViewController {
-        let controller = UIStoryboard(name: "Home", bundle: nil).instantiateViewController(withIdentifier: "HomeViewController") as! HomeViewController
-        controller.active = active
-        return controller
+
+    var frame: CGRect!
+
+    static func loadFromStoryboard() -> HomeViewController {
+        return UIStoryboard(name: "Home", bundle: nil).instantiateViewController(withIdentifier: "HomeViewController") as! HomeViewController
     }
     
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        frame = view.frame
+        
+        updateInfoView()
+        activateOmniBar()
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(HomeViewController.onKeyboardChangeFrame), name: .UIKeyboardWillChangeFrame, object: nil)
+    }
+
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        if active {
-            enterActiveMode()
-        } else {
-            enterPassiveMode()
+        
+        let feature = HomeRowOnboardingFeature()
+        if !feature.showNow() {
+            hideCallToAction()
         }
-    }
     
-    @IBAction func onEnterActiveModeTapped(_ sender: Any) {
-         enterActiveModeAnimated()
     }
     
-    private func enterActiveModeAnimated() {
-        UIView.animate(withDuration: Constants.animationDuration, animations: {
-            self.moveSearchBarUp()
-        }, completion: { _ in
-            self.enterActiveMode()
-        })
-    }
-    
-    private func enterActiveMode() {
-        chromeDelegate?.setNavigationBarHidden(false)
-        passiveContent.isHidden = true
-        delegate?.homeDidActivateOmniBar(home: self)
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        chromeDelegate?.omniBar.becomeFirstResponder()
     }
 
-    @IBAction func onEnterPassiveModeTapped(_ sender: Any) {
-        enterPassiveModeAnimated()
+    @IBAction func hideKeyboard() {
+        chromeDelegate?.omniBar.resignFirstResponder()
     }
     
-    private func enterPassiveModeAnimated() {
-        chromeDelegate?.setNavigationBarHidden(true)
-        passiveContent.isHidden = false
-        UIView.animate(withDuration: Constants.animationDuration, animations: {
```

Comment:
You can delete the `Constants.animationDuration` constant too now

Reply:
Done